lisa/sut_orchestrator/util/device_pool.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, List, Optional
+
+from lisa.sut_orchestrator.util.schema import HostDevicePoolSchema, HostDevicePoolType
+from lisa.util import LisaException, SkippedException
+
+
+class BaseDevicePoolImpl:
```

Comment:
@cwize1 

There are common functionalities between 2 platform when it comes to device pool implementation like creating device pool, get/put devices from/into the pool, getting primary NIC etc.

But the implementation is platform specific as hyperv/libvirt has some fundamental differences when it comes to device passthrough implementation. Due to this nature of the requirement, It seems having proper parent class for device pool make sense here. This would even help to eliminate duplicate code for device pool functionality between the SUT platforms.